src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -990,9 +959,48 @@ internal Run CreateRun(RunVersionOne v1Run)
                 }
             }
 
+            _currentRun = null;
+
             return run;
         }
 
+        private void PopulateLogicalLocation(
+            Run v2Run, 
+            IDictionary<string, LogicalLocationVersionOne> v1LogicalLocations, 
+            string fullyQualifiedName, 
+            LogicalLocationVersionOne v1LogicalLocation, 
+            HashSet<string> populatedKeys)
+        {
+            // We saw and populated this one previously, because it was a parent to 
+            // a logical location that we encountered earlier
+            if (populatedKeys.Contains(fullyQualifiedName)) { return; }
```

Comment:
>fullyQualifiedName [](start = 39, length = 18)

Given how you called this method on Line 920, it's clear that the thing you're calling FQLN here is actually the v1 dictionary key. So, given the massive potential for confusion between the key and the FQLN, manifested several places above, it's important to name this `v1Key`, and make sure not to use it in a place where you really do need the FQLN.